sway/config/output.c

```diff
@@ -277,17 +277,49 @@ void apply_output_config(struct output_config *oc, struct sway_output *output) {
 }
 
 static struct output_config *get_output_config(char *name, char *identifier) {
+	struct output_config *oc_name = NULL;
 	int i = list_seq_find(config->output_configs, output_name_cmp, name);
 	if (i >= 0) {
-		return config->output_configs->items[i];
+		oc_name = config->output_configs->items[i];
 	}
 
+	struct output_config *oc_id = NULL;
 	i = list_seq_find(config->output_configs, output_name_cmp, identifier);
 	if (i >= 0) {
-		return config->output_configs->items[i];
+		oc_id = config->output_configs->items[i];
 	}
 
-	return NULL;
+	struct output_config *result = NULL;
+	if (oc_name && oc_id) {
+		// Generate a config named `<identifier> on <name>` which contains a
+		// merged copy of the identifier on name. This will make sure that both
+		// identifier and name configs are respected, with identifier getting
+		// priority
+		size_t length = strlen(identifier) + strlen(" on ") + strlen(name) + 1;
+		char *temp = malloc(length);
```

Comment:
Need to `free` this. Also, when computing the length it'd probably be better to use `snprintf`'s return value instead of `strlen`.

Reply:
Fixed